application.py

```diff
@@ -276,6 +299,30 @@ def confirm_register(token):
 
     return redirect("/problem/helloworld")
 
+@app.route('/confirmlogin/<token>')
+def confirm_login(token):
+    try:
+        token = jwt.decode(token, app.config['SECRET_KEY'], algorithms=['HS256'])
+    except Exception as e:
+        sys.stderr.write(str(e))
+        token = 0
+    if not token:
+        return render_template("login.html",
+                               message="Login verification link invalid",
+                               site_key=app.config['HCAPTCHA_SITE'])
+    if datetime.strptime(token["expiration"], "%Y-%m-%dT%H:%M:%S.%f") < datetime.utcnow():
+        return render_template("login.html",
+                               message="Login verification link expired; Please re-login",
+                               site_key=app.config['HCAPTCHA_SITE'])
+
+    # Log user in
+    user = db.execute(
+        "SELECT * FROM users WHERE email = :email", email=token['email'])[0]
+    session["user_id"] = user["id"]
+    session["username"] = user["username"]
+    session["admin"] = False
+
+    return redirect("/problem/helloworld")
```

Comment:
should redirect to / after logging in. People are only redirected to /problem/helloworld after registering